File: android/src/main/java/kz/mybpm/core/multiple/fcm/FirebaseConfig.java
package kz.mybpm.core.multiple.fcm;

public class FirebaseConfig {

  public static final String FC_APPLICATION_NAME = "firebaseAppName";
  public static final String FC_PROJECT_ID = "projectId";
  public static final String FC_APPLICATION_ID = "applicationId";
  public static final String FC_API_KEY = "apiKey";
  public static final String FC_GCM_SENDER_ID = "gcmSenderId";
  public static final String FC_STORAGE_BUCKET = "storageBucket";
  public static final String FC_DATABASE_URL = "databaseUrl";
  public static final String FC_GA_TRACKING_ID = "gaTrackingId";



  private String firebaseAppName;
  private String apiKey;
  private String applicationId;
  private String databaseUrl;
  private String gaTrackingId;
  private String gcmSenderId;
  private String storageBucket;
  private String projectId;


  public FirebaseConfig() {

  }

  public FirebaseConfig(String firebaseAppName, String apiKey, String applicationId, String databaseUrl,
                        String gaTrackingId, String gcmSenderId, String storageBucket, String projectId) {
    this.firebaseAppName = firebaseAppName;
    this.apiKey = apiKey;
    this.applicationId = applicationId;
    this.databaseUrl = databaseUrl;
    this.gaTrackingId = gaTrackingId;
    this.gcmSenderId = gcmSenderId;
    this.storageBucket = storageBucket;
    this.projectId = projectId;
  }

  public String getFirebaseAppName() {
    return firebaseAppName;
  }

  public FirebaseConfig setFirebaseAppName(String firebaseAppName) {
    this.firebaseAppName = firebaseAppName;
    return this;
  }

  public String getApiKey() {
    return apiKey;
  }

  public FirebaseConfig setApiKey(String apiKey) {
    this.apiKey = apiKey;
    return this;
  }

  public String getApplicationId() {
    return applicationId;
  }

  public FirebaseConfig setApplicationId(String applicationId) {
    this.applicationId = applicationId;
    return this;
  }

  public String getDatabaseUrl() {
    return databaseUrl;
  }

  public FirebaseConfig setDatabaseUrl(String databaseUrl) {
    this.databaseUrl = databaseUrl;
    return this;
  }

  public String getGaTrackingId() {
    return gaTrackingId;
  }

  public FirebaseConfig setGaTrackingId(String gaTrackingId) {
    this.gaTrackingId = gaTrackingId;
    return this;
  }

  public String getGcmSenderId() {
    return gcmSenderId;
  }

  public FirebaseConfig setGcmSenderId(String gcmSenderId) {
    this.gcmSenderId = gcmSenderId;
    return this;
  }

  public String getStorageBucket() {
    return storageBucket;
  }

  public FirebaseConfig setStorageBucket(String storageBucket) {
    this.storageBucket = storageBucket;
    return this;
  }

  public String getProjectId() {
    return projectId;
  }

  public FirebaseConfig setProjectId(String projectId) {
    this.projectId = projectId;
    return this;
  }


}

File: ios/Plugin/FirebaseConfig.swift
import Foundation

class FirebaseConfig: Codable
{
    static var FC_APPLICATION_NAME : String = "firebaseAppName";
    static var FC_PROJECT_ID : String = "projectId";
    static var FC_APPLICATION_ID : String = "applicationId";
    static var FC_API_KEY : String = "apiKey";
    static var FC_GCM_SENDER_ID : String = "gcmSenderId";
    static var FC_STORAGE_BUCKET : String = "storageBucket";
    static var FC_DATABASE_URL : String = "databaseUrl";
    static var FC_GA_TRACKING_ID : String = "gaTrackingId";


    private var firebaseAppName : String;
    private var apiKey : String;
    private var applicationId : String;
    private var databaseUrl : String;
    private var gaTrackingId : String;
    private var gcmSenderId : String;
    private var storageBucket : String;
    private var projectId : String;


    init(_ firebaseAppName : String, _ apiKey : String, _ applicationId : String, _ databaseUrl : String, _ gaTrackingId : String, _ gcmSenderId : String, _ storageBucket : String, _ projectId : String)
    {
        self.firebaseAppName = firebaseAppName;
        self.apiKey = apiKey;
        self.applicationId = applicationId;
        self.databaseUrl = databaseUrl;
        self.gaTrackingId = gaTrackingId;
        self.gcmSenderId = gcmSenderId;
        self.storageBucket = storageBucket;
        self.projectId = projectId;
    }
    func getFirebaseAppName() -> String?
    {
        return self.firebaseAppName;
    }
    func setFirebaseAppName(_ firebaseAppName : String) -> FirebaseConfig?
    {
        self.firebaseAppName = firebaseAppName;
        return self;
    }
    func getApiKey() -> String?
    {
        return self.apiKey;
    }
    func setApiKey(_ apiKey : String) -> FirebaseConfig?
    {
        self.apiKey = apiKey;
        return self;
    }
    func getApplicationId() -> String?
    {
        return self.applicationId;
    }
    func setApplicationId(_ applicationId : String) -> FirebaseConfig?
    {
        self.applicationId = applicationId;
        return self;
    }
    func getDatabaseUrl() -> String?
    {
        return self.databaseUrl;
    }
    func setDatabaseUrl(_ databaseUrl : String) -> FirebaseConfig?
    {
        self.databaseUrl = databaseUrl;
        return self;
    }
    func getGaTrackingId() -> String?
    {
        return self.gaTrackingId;
    }
    func setGaTrackingId(_ gaTrackingId : String) -> FirebaseConfig?
    {
        self.gaTrackingId = gaTrackingId;
        return self;
    }
    func getGcmSenderId() -> String?
    {
        return self.gcmSenderId;
    }
    func setGcmSenderId(_ gcmSenderId : String) -> FirebaseConfig?
    {
        self.gcmSenderId = gcmSenderId;
        return self;
    }
    func getStorageBucket() -> String?
    {
        return self.storageBucket;
    }
    func setStorageBucket(_ storageBucket : String) -> FirebaseConfig?
    {
        self.storageBucket = storageBucket;
        return self;
    }
    func getProjectId() -> String?
    {
        return self.projectId;
    }
    func setProjectId(_ projectId : String) -> FirebaseConfig?
    {
        self.projectId = projectId;
        return self;
    }
}
